airbyte-integrations/connectors/source-amazon-seller-partner/unit_tests/integration/test_report_based_streams.py

```diff
@@ -0,0 +1,266 @@
+
+
+def _get_document_download_url_response(compressed: Optional[bool] = False) -> HttpResponse:
+    response_body = {"reportDocumentId": _REPORT_DOCUMENT_ID, "url": _DOCUMENT_DOWNLOAD_URL}
+    if compressed:
+        # See https://developer-docs.amazon.com/sp-api/docs/reports-api-v2021-06-30-reference#compressionalgorithm
+        response_body["compressionAlgorithm"] = "GZIP"
+    return build_response(response_body, status_code=200)
+
+
+def _download_document_response(stream_name: str, data_format: Optional[str] = "csv", compressed: Optional[bool] = False) -> HttpResponse:
+    response_body = find_template(stream_name, __file__, data_format)
+    if compressed:
+        response_body = gzip.compress(response_body.encode("iso-8859-1"))
+    return HttpResponse(body=response_body, status_code=200)
+
+
+@freezegun.freeze_time(_NOW.isoformat())
+class TestFullRefresh(TestCase):
+
+    @staticmethod
+    def _read(stream_name: str, config_: ConfigBuilder, expecting_exception: bool = False) -> EntrypointOutput:
+        return read_output(config_, stream_name, SyncMode.full_refresh, expecting_exception=expecting_exception)
+
+    @HttpMocker()
+    def test_given_report_when_read_then_return_records(self, http_mocker: HttpMocker) -> None:
+        for stream_name, params in STREAMS.items():
+            with self.subTest(msg=stream_name):
+                _mock_auth(http_mocker)
+
+                http_mocker.post(_create_report_request(stream_name).build(), _create_report_response())
+                http_mocker.get(_check_report_status_request(_REPORT_ID).build(), _check_report_status_response(stream_name))
+                http_mocker.get(
+                    _get_document_download_url_request(_REPORT_DOCUMENT_ID).build(), _get_document_download_url_response()
+                )
+                http_mocker.get(
+                    _download_document_request(_DOCUMENT_DOWNLOAD_URL).build(),
+                    _download_document_response(stream_name, data_format=params.get("data_format", "csv")),
+                )
+
+                output = self._read(stream_name, config())
+                assert len(output.records) == 2
+
+    @HttpMocker()
+    def test_given_compressed_report_when_read_then_return_records(self, http_mocker: HttpMocker) -> None:
+        for stream_name, params in STREAMS.items():
+            with self.subTest(msg=stream_name):
+                _mock_auth(http_mocker)
+
+                http_mocker.post(_create_report_request(stream_name).build(), _create_report_response())
+                http_mocker.get(_check_report_status_request(_REPORT_ID).build(), _check_report_status_response(stream_name))
+                http_mocker.get(
+                    _get_document_download_url_request(_REPORT_DOCUMENT_ID).build(),_get_document_download_url_response(compressed=True)
+                )
+
+                # a workaround to pass compressed document to the mocked response
```

Comment:
I've created [an issue](https://github.com/airbytehq/airbyte/issues/34558) for that